lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb

```diff
@@ -7,341 +7,230 @@ module Providers
+            gene = record.raw_fields['gene']
+            variant = record.raw_fields['genotype']
+            teststatus = record.raw_fields['teststatus']
+            if gene.present? && variant.present? && pathogenic?(record)
               genocolorectal.add_status(2)
-              genocolorectal.add_zygosity(geno)
-            when /fail/
+            elsif gene.present? && variant.blank?
+              genocolorectal.add_status(4)
+            elsif teststatus.present? && teststatus.scan(/fail/i).size.positive?
               genocolorectal.add_status(9)
-            when nil
-              genocolorectal.add_status(1)
             else
-              @logger.info "Encountered unfamiliar teststatus string: #{geno}"
+              genocolorectal.add_status(1)
             end
           end
 
-          def add_cdna_change_from_report(genocolorectal, record)
-            case record.raw_fields['genotype']
-            when CDNA_REGEX
-              genocolorectal.add_gene_location($LAST_MATCH_INFO[:cdna])
-              @logger.debug "SUCCESSFUL cdna change parse for: #{$LAST_MATCH_INFO[:cdna]}"
+          def process_variant_records(genocolorectal, record)
+            genocolorectals = []
+            if full_screen?(genocolorectal)
+              investigation_code = record.raw_fields['investigation code']&.downcase&.strip
+              @genes_panel = INVESTIGATION_CODE_GENE_MAPPING[investigation_code]
+              process_fullscreen_records(genocolorectal, record, genocolorectals)
+            elsif targeted?(genocolorectal) || no_scope?(genocolorectal)
+              process_targeted_screen(genocolorectal, record, genocolorectals)
             end
+            genocolorectals
           end
 
-          def process_protein_impact(genocolorectal, record)
-            if IMPACT_REGEX.match(record.raw_fields['genotype'])
-              genocolorectal.add_protein_impact($LAST_MATCH_INFO[:impact])
-              @logger.debug "SUCCESSFUL PROTEIN change parse for: #{$LAST_MATCH_INFO[:impact]}"
-            else
-              @logger.debug "UNSUCCESSFUL PROTEIN PARSE FOR #{record.raw_fields['genotype']}"
+          def process_fullscreen_records(genocolorectal, record, genocolorectals)
+            variant = record.raw_fields['genotype']
+            gene = get_gene(record)
+            genocolorectal.add_gene_colorectal(gene)
+            if positive_rec?(genocolorectal)
+              add_fs_negative_genes(gene, genocolorectal, genocolorectals, record)
+              process_variants(genocolorectal, variant)
+              genocolorectals.append(genocolorectal)
+            elsif gene.present? # for other status records
+              add_fs_negative_genes(gene, genocolorectal, genocolorectals, record)
+              genocolorectals.append(genocolorectal)
+            else # for other status null gene record
+              process_null_gene_rec(genocolorectal, genocolorectals)
             end
+
+            genocolorectals
```

Comment:
@shilpigoeldev not sure you need to return `genocolorectals` here. The result of this method isn't assigned anywhere, the method mutates `genocolorectals`

Reply:
Done , on all the methods where not assigning to another variable.